test/unit_tests/braket/circuits/test_gate_calibration.py

```diff
@@ -0,0 +1,148 @@
+    )
+
+
+def test_gc_creation(pulse_sequence):
+    calibration_key = (Gate.H(), QubitSet([0, 1]))
+    calibration = GateCalibrations({calibration_key: pulse_sequence})
+
+    assert calibration.pulse_sequences[calibration_key] == pulse_sequence
+
+
+def test_gc_copy(pulse_sequence):
+    calibration_key = (Gate.H(), QubitSet([0, 1]))
+    calibration = GateCalibrations({calibration_key: pulse_sequence})
+
+    assert calibration == calibration.copy()
+
+
+def test_filter(pulse_sequence):
+    calibration_key = (Gate.Z(), QubitSet([0, 1]))
+    calibration_key_2 = (Gate.H(), QubitSet([0, 1]))
+    calibration = GateCalibrations(
+        {calibration_key: pulse_sequence, calibration_key_2: pulse_sequence}
+    )
+    expected_calibration_1 = GateCalibrations({calibration_key: pulse_sequence})
+    expected_calibration_2 = GateCalibrations(
+        {calibration_key: pulse_sequence, calibration_key_2: pulse_sequence}
+    )
+    expected_calibration_3 = GateCalibrations({calibration_key_2: pulse_sequence})
+    assert expected_calibration_1 == calibration.filter(gates=[Gate.Z()])
+    assert expected_calibration_2 == calibration.filter(qubits=QubitSet(0))
+    assert expected_calibration_3 == calibration.filter(gates=[Gate.H()], qubits=QubitSet(1))
+
+
+def test_to_ir(pulse_sequence):
+    calibration_key = (Gate.Rx(angle=1), QubitSet([0, 1]))
+    calibration = GateCalibrations({calibration_key: pulse_sequence})
+    expected_ir = "\n".join(
+        [
+            "OPENQASM 3.0;",
+            "defcal rx(1.0) $0, $1 {",
+            "    barrier test_frame_rf;",
+            "    delay[1000000000000.0ns] test_frame_rf;",
+            "}",
+        ]
+    )
+
+    assert calibration.to_ir() == expected_ir
+
+
+@pytest.mark.xfail(raises=ValueError)
+def test_to_ir_with_bad_key(pulse_sequence):
+    calibration_key = (Gate.Z(), QubitSet([0, 1]))
+    calibration_key_2 = (Gate.H(), QubitSet([0, 1]))
+    calibration = GateCalibrations(
+        {calibration_key: pulse_sequence, calibration_key_2: pulse_sequence}
+    )
+    expected_ir = "\n".join(
+        [
+            "OPENQASM 3.0;",
+            "defcal z $0, $1 {",
+            "    barrier test_frame_rf;",
+            "    delay[1000000000000.0ns] test_frame_rf;",
+            "}",
+        ]
+    )
+    assert expected_ir == calibration.to_ir((Gate.Z(), QubitSet([1, 2])))
+
+
+def test_to_ir_with_key(pulse_sequence):
+    calibration_key = (Gate.Z(), QubitSet([0, 1]))
+    calibration_key_2 = (Gate.H(), QubitSet([0, 1]))
+    calibration = GateCalibrations(
+        {calibration_key: pulse_sequence, calibration_key_2: pulse_sequence}
+    )
+    expected_ir = "\n".join(
+        [
+            "OPENQASM 3.0;",
+            "defcal z $0, $1 {",
+            "    barrier test_frame_rf;",
+            "    delay[1000000000000.0ns] test_frame_rf;",
+            "}",
+        ]
+    )
+    assert expected_ir == calibration.to_ir(calibration_key)
+
+
+def test_ngc_length(pulse_sequence):
```

Comment:
Please rename all ngc references in this PR to gate calibrations since ngc will be very confusing for someone else looking at this code.

Reply:
Fixed